internal/kafka/internal/workers/kafka_mgrs/kafkas_mgr.go

```diff
@@ -120,6 +125,13 @@ func (k *KafkaManager) Reconcile() []error {
 		return encounteredErrors
 	}
 
+	// reconciles expires_at field for kafka instances
+	updateExpiresAtErrors := k.reconcileKafkaExpiresAt(kafkas)
```

Comment:
Here there is a potential race condition where `updateZeroValueOfKafkaRequestsExpiredAt` updates the value of some kafka instances in the DB but the `kafkas` slice in memory does not contain the changes

Reply:
do you think it's worth calling ListAll within this method again to avoid that?

Reply:
With the current logic we might get a scenario where the ExpiresAt of some of the kafka instances would be the zero-value of time.Time which is 1 Jan of year 1. The `Valid` attribute would also be set to `true`.

I think this might have an impact on the logic that marks instances as suspended cc @ziccardi.

Regarding the new logic, it also references ExpiresAt but I am not fully sure if it might cause any issue. Could you review it? I think we should be ok from that side but more people double checking it would be useful

Reply:
I double checked the suspension logic to see how it would be affected if the expires_at has a zero time value. Looks like it should be ok because of this check:

https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/b4e39ab567773b7446925d67c4f4905e9ddb5cb4/internal/kafka/internal/workers/kafka_mgrs/kafkas_mgr.go#L129-L133

The Kafka will be deemed as 'expired' because `Valid` will be true but it won't get suspended.

The problem lies in the deprovision logic because it doesn't account for a zero time value if by chance a Kafka instance expiration date hasn't been updated at this point.
https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/b4e39ab567773b7446925d67c4f4905e9ddb5cb4/internal/kafka/internal/services/kafka.go#L647-L657

I think we can add a check here to accommodate for that until this transition to the null value is complete. wdyt? @miguelsorianod @ziccardi

Reply:
I think there shouldn't be any issue because the Kafkas are retrieved from the DB in `DeprovisionExpiredKafkas` right?
If we want we can still be extra sure and temporarily add what you said to avoid any surprise during 2 pods running at the same time due to some race condition. I can't remember if that's even possible at the moment

Reply:
> I think there shouldn't be any issue because the Kafkas are retrieved from the DB in DeprovisionExpiredKafkas right?

Yeah i don't think there will be any issues. As far as I know there can only be one leader pod so it shouldn't be possible.